server/apps/research/views.py

```
@@ -238,6 +239,62 @@ def categories(self, request):
                 {'success': False, 'error': 'An error occurred while fetching categories'},
                 status=status.HTTP_500_INTERNAL_SERVER_ERROR
             )
+    # Custom action to retrieve articles by primary category
+    @action(detail=False, methods=['get'], url_path=r'primary-category/(?P<category_slug>[-\w]+)')
+    def retrieve_by_primary_category(self, request, category_slug=None):
```

Ruff (0.8.2)

Redefinition of unused `retrieve_by_primary_category` from line 139

(F811)